### ISSUE TITLE
Failed unmarshalling using circe

### ISSUE THREAD

**Issue by ssuarez6** [howardjohn/scala-server-lambda#3]
Say you define a Route:
```
object Lambda{
  val route: Route = path("sample"){
    post{
      entity(as[MyClass]){ entity =>
        complete(entity)
      }
    }
  }
}
```

using [generic marshallers and unmarshallers](https://github.com/hseeberger/akka-http-json) and checking circe versions, it will return `415 Unsupported media type`

**Comment by howardjohn** [comment]
Thanks for bringing this up. Basically what the issue is is that the content types are not passed through to akka, so everything is counted as a string. The generic marshallers and unmarshallers are strict and will only convert http bodies with a json content type. I will look into fixing this in a bit, I think I know how I can.

For now, if you change your (un)marshallers to this:
```scala
  implicit final def unmarshaller[A: Decoder]: FromEntityUnmarshaller[A] =
    Unmarshaller.stringUnmarshaller
     // .forContentTypes(`application/json`) // This is the line that causes the issue
      .flatMap { ctx => mat => json =>
        decode[A](json).fold(Future.failed, Future.successful)
      }

  implicit final def marshaller[A: Encoder]: ToEntityMarshaller[A] =
    Marshaller.withFixedContentType(`application/json`) { a =>
      HttpEntity(`application/json`, a.asJson.noSpaces)
    }
```
It should work. Note the commented line, which is what was causing this error.

I don't think this affects the http4s one, but I will check as well.

**Comment by ssuarez6** [comment]
Notice this only happens when unmarshalling, custom marshallers work correctly.
Let me know if any help is needed :smile:

**Comment by howardjohn** [comment]
Should be fixed now from https://github.com/howardjohn/scala-server-lambda/commit/baf3ddacc3adaee9981598f22fcde8d311eb4135. 

I published the new version, `0.4.0-SNAPSHOT`

Let me know if there are any issues!
